File: lib/division_parser.rb
require 'nokogiri'
require 'open-uri'
require 'pry'
require 'json'

html = open('http://www.police.gov.hk/ppp_en/contact_us.html').read
doc = Nokogiri::HTML(html)

output = doc
  .search('#main_content_area > table.table_01')
  .select { |e| # only section with header
    e.search('.sub_header').count > 0
  }
  .select { |e| # only supported sections
    title = e.search('.sub_header').text.strip
    [
      "HONG KONG ISLAND",
      "KOWLOON EAST",
      "KOWLOON WEST",
      "NEW TERRITORIES SOUTH",
      "NEW TERRITORIES NORTH",
      "MARINE"
    ].include?(title)
  }
  .map { |e|
    region = e.search('.sub_header').text.strip
    tr = e.search('./tr')
    divisions = tr
      .select { |tr| tr.search('./td').count == 3 }
      .map{|tr| tr.search('td:nth-child(1)').children[0].text.strip }
    { region: region, divisions: divisions }
  }

puts JSON.pretty_generate(output)

File: lib/division_parser.js
var fs = require('fs')
var cheerio = require('cheerio')
var request = require('request')
var _ = require('lodash')

request('http://www.police.gov.hk/ppp_en/contact_us.html', (req, res) => {
  var $ = cheerio.load(res.body)
  var tables = $('#main_content_area table.table_01')

  var districts = tables
    .filter((t) => $('.sub_header', t).length > 0)
    .map((e) => {
      var region = $('.sub_header', e).text()
      var divisions = $('tr', e)
        .filter((tr) => { $('td', tr).length == 3 })
        .map((tr) => { $('td:nth-child(1)', tr).children[0].text() })
      return {
        region: region,
        divisions: divisions
      }
    })

  console.log("districts ", districts)
})
